provider/aws/aws.go

```diff
@@ -416,6 +416,9 @@ func (p *AWSProvider) records(ctx context.Context, zones map[string]*route53.Hos
 			}
 
 			for _, ep := range newEndpoints {
+				if ep == nil {
+					continue
+				}
```

Comment:
Mmmmh :thinking: 
Wdyt about avoiding to append it when it's nil ? 

Something like this on line 399

```go
				ep := endpoint.NewEndpointWithTTL(wildcardUnescape(aws.StringValue(r.Name)), aws.StringValue(r.Type), ttl, targets...)
                if ep != nil {
				  if aws.StringValue(r.Type) == endpoint.RecordTypeCNAME {
					ep = ep.WithProviderSpecific(providerSpecificAlias, "false")
				  }
				  newEndpoints = append(newEndpoints, ep)
                }
```

And add a nil check also on line 412:

```go
				ep := endpoint.
					NewEndpointWithTTL(wildcardUnescape(aws.StringValue(r.Name)), endpoint.RecordTypeA, ttl, aws.StringValue(r.AliasTarget.DNSName))
                if ep != nil {
				  ep = ep.WithProviderSpecific(providerSpecificEvaluateTargetHealth, fmt.Sprintf("%t", aws.BoolValue(r.AliasTarget.EvaluateTargetHealth))).
					  WithProviderSpecific(providerSpecificAlias, "true")
				  newEndpoints = append(newEndpoints, ep)
                }
```